src/modules/users/users.service.ts

```diff
@@ -1,28 +1,129 @@
+    });
+
+    return users;
   }
 
-  findAll() {
-    return `This action returns all user`;
+  async getProfile(accountId: string) {
+    const account = await this.userAccountRepository.findOne({
+      where: { id: accountId, isDeleted: false },
+      relations: ['profile'],
+    });
+    if (!account || !account.profile) {
+      throw new NotFoundException('Account not found');
+    }
+
+    return account.profile;
+  }
+
+  async getUserWishlist(accountId: string) {
+    const account = await this.userAccountRepository.findOne({
+      where: { id: accountId, isDeleted: false },
+    });
+    if (!account) {
+      throw new NotFoundException('Account not found');
+    }
+
+    const userWishlist = await this.wishlistRepository.find({
+      where: {
+        accountId,
+      },
+    });
+
+    return userWishlist;
+  }
+
+  async updateProfile(accountId: string, updateProfileDto: UpdateProfileDto) {
+    const account = await this.userAccountRepository.findOne({
+      where: { id: accountId, isDeleted: false },
+      relations: ['profile'],
+    });
+    if (!account || !account.profile) {
+      throw new NotFoundException('Account not found');
+    }
+
+    Object.assign(account.profile, updateProfileDto);
+
+    const savedProfile = await this.userProfileRepository.save(account.profile);
+    return savedProfile;
   }
 
-  findOne() {
-    return `This action returns user`;
+  async updateProfileAvatar(accountId: string, avatarUrl: string) {
+    const account = await this.userAccountRepository.findOne({
+      where: { id: accountId, isDeleted: false },
+      relations: ['profile'],
+    });
+    if (!account || !account.profile) {
+      throw new NotFoundException('Account not found');
+    }
+
+    account.profile.image = avatarUrl;
+
+    const savedProfile = await this.userProfileRepository.save(account.profile);
+    return savedProfile;
   }
 
-  update() {
-    return `This action updates user`;
+  async toggleUserRole(accountId: string) {
+    const account = await this.userAccountRepository.findOne({
+      where: { id: accountId, isDeleted: false },
+    });
+    if (!account) {
+      throw new NotFoundException('Account not found');
+    }
+
+    account.role = account.role === Roles.Admin ? Roles.User : Roles.Admin;
+
+    const updatedUser = await this.userAccountRepository.save(account);
+    return updatedUser;
   }
 
-  remove() {
-    return `This action removes user`;
+  async deleteAccount(deleterId: string, deletingAccountId: string) {
+    const deleter = await this.userAccountRepository.findOne({
+      where: { id: deleterId, isDeleted: false },
+    });
+    if (deleter.role != Roles.Admin && deleterId != deletingAccountId) {
+      throw new ForbiddenException('Only owner or admin can delete account');
+    }
+
+    const deletingUser = await this.userAccountRepository.findOne({
+      where: { id: deletingAccountId, isDeleted: false },
+    });
```

Comment:
It should be `isDeleted: true` for this to work correctly.
You don't have to search the account if deleteActor and deletingAccount are the same, I suppose.